src/helpers/clientChronicles.js

```diff
@@ -0,0 +1,193 @@
+import axios from 'axios';
+import {ulid} from 'ulid'
+
+const CHRONICLES_BACKEND = process.env.REACT_APP_CHRONICLES_BACKEND;
+const chroniclesUrl = path => `${CHRONICLES_BACKEND}${path}`;
+
+//An array of DOM events that should be interpreted as user activity.
+const ACTIVITY_EVENTS = ['mousedown', 'mousemove', 'keydown', 'scroll', 'touchstart'];
+
+const FLOWS = [
+  {start: 'page-enter',               end: 'page-leave',                 subFlows: []},
+  {start: 'unit-page-enter',          end: 'unit-page-leave',            subFlows: ['player-play']},
+  {start: 'collection-page-enter',    end: 'collection-page-leave',      subFlows: ['collection-unit-selected']},
+  {start: 'collection-unit-selected', end: 'collection-unit-unselected', subFlows: ['player-play']},
+  {start: 'player-play',              end: 'player-stop',                subFlows: []},
+];
+
+const FLOWS_BY_END = new Map(FLOWS.map(flow => [flow.end, flow]));
+const SUBFLOWS = new Map(Object.entries(FLOWS.reduce((acc, flow) => {
+  flow.subFlows.forEach(subflow => {
+    if (!(subflow in acc)) {
+      acc[subflow] = [];
+    }
+    if (!acc[subflow].includes(flow.start)) {
+      acc[subflow].push(flow.start);
+    }
+  });
+  return acc;
+}, {})));
+
+const MAX_INACTIVITY_MS = 60 * 1000; // Minute in milliseconds.
+
+export default class ClientChronicles {
+  constructor(history) {
+    if (!CHRONICLES_BACKEND) {
+      return;
```

Comment:
Maybe to replace here append() to a noop? Then you don't need to care about it later.

Reply:
Done.